src/main/java/org/jetbrains/research/kotlinrminer/decomposition/OperationInvocation.java

```diff
@@ -0,0 +1,522 @@
+        String invocationAsString = creation.toString();
+        String suffix = invocationAsString
+            .substring(expressionAsString.length() + 1, invocationAsString.length());
+        subExpressions.add(0, suffix);
+        processExpression(creation.getBody(), subExpressions);
+      } else {
+        subExpressions.add(0, creation.toString());
+      }
+    }
+  }
+
+  public OperationInvocation update(String oldExpression, String newExpression) {
+    OperationInvocation newOperationInvocation = new OperationInvocation();
+    newOperationInvocation.methodName = this.methodName;
+    newOperationInvocation.locationInfo = this.locationInfo;
+    update(newOperationInvocation, oldExpression, newExpression);
+    newOperationInvocation.subExpressions = new ArrayList<String>();
+    for (String argument : this.subExpressions) {
+      newOperationInvocation.subExpressions.add(
+          ReplacementUtil.performReplacement(argument, oldExpression, newExpression));
+    }
+    return newOperationInvocation;
+  }
+
+  public String getName() {
+    return getMethodName();
+  }
+
+  public String getMethodName() {
+    return methodName;
+  }
+
+  public int numberOfSubExpressions() {
+    return subExpressions.size();
+  }
+
+  public boolean matchesOperation(UMLOperation operation) {
+    return matchesOperation(operation, new HashMap<String, UMLType>(), null);
+  }
+
+  public boolean matchesOperation(UMLOperation operation, Map<String, UMLType> variableTypeMap,
+                                  UMLModelDiff modelDiff) {
+    List<UMLType> inferredArgumentTypes = new ArrayList<UMLType>();
+    for (String arg : arguments) {
+      int indexOfOpeningParenthesis = arg.indexOf("(");
+      int indexOfOpeningSquareBracket = arg.indexOf("[");
+      boolean openingParenthesisBeforeSquareBracket = false;
+      boolean openingSquareBracketBeforeParenthesis = false;
+      if (indexOfOpeningParenthesis != -1 && indexOfOpeningSquareBracket != -1) {
+        if (indexOfOpeningParenthesis < indexOfOpeningSquareBracket) {
+          openingParenthesisBeforeSquareBracket = true;
+        } else if (indexOfOpeningSquareBracket < indexOfOpeningParenthesis) {
+          openingSquareBracketBeforeParenthesis = true;
+        }
+      } else if (indexOfOpeningParenthesis != -1 && indexOfOpeningSquareBracket == -1) {
+        openingParenthesisBeforeSquareBracket = true;
+      } else if (indexOfOpeningParenthesis == -1 && indexOfOpeningSquareBracket != -1) {
+        openingSquareBracketBeforeParenthesis = true;
+      }
+      if (variableTypeMap.containsKey(arg)) {
+        inferredArgumentTypes.add(variableTypeMap.get(arg));
+      } else if (arg.startsWith("\"") && arg.endsWith("\"")) {
+        inferredArgumentTypes.add(UMLType.extractTypeObject("String"));
+      } else if (arg.startsWith("\'") && arg.endsWith("\'")) {
+        inferredArgumentTypes.add(UMLType.extractTypeObject("char"));
```

Comment:
In Kotlin, names of types start with an upper case letter. We should use `Char`, `Boolean` here